provider/vsphere/environ_network.go

```diff
@@ -9,23 +9,29 @@ import (
 	"github.com/juju/juju/network"
 )
 
-// OpenPorts opens the given port ranges for the whole environment.
-// Must only be used if the environment was setup with the
-// FwGlobal firewall mode.
-func (env *environ) OpenPorts(rules []network.IngressRule) error {
+// OpenPorts is part of the environs.Firewaller interface.
+func (*environ) OpenPorts(rules []network.IngressRule) error {
 	return errors.Trace(errors.NotSupportedf("ClosePorts"))
 }
 
-// ClosePorts closes the given port ranges for the whole environment.
-// Must only be used if the environment was setup with the
-// FwGlobal firewall mode.
-func (env *environ) ClosePorts(rules []network.IngressRule) error {
+func (*sessionEnviron) OpenPorts(rules []network.IngressRule) error {
 	return errors.Trace(errors.NotSupportedf("ClosePorts"))
 }
 
-// IngressRules returns the port ranges opened for the whole environment.
-// Must only be used if the environment was setup with the
-// FwGlobal firewall mode.
-func (env *environ) IngressRules() ([]network.IngressRule, error) {
+// ClosePorts is part of the environs.Firewaller interface.
+func (*environ) ClosePorts(rules []network.IngressRule) error {
+	return errors.Trace(errors.NotSupportedf("ClosePorts"))
+}
+
+func (*sessionEnviron) ClosePorts(rules []network.IngressRule) error {
```

Comment:
I did vacillate a bit on this, and thought that embedding might be too dangerous. As long as we have reasonable test coverage, it should be fine, so I'll do that.